.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: ci
+
+on:
+  push:
+    branches:
+      - "master"
+  pull_request: {}
+
+jobs:
+  build:
+    name: "CI"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build Docker image
+        run: |
+          docker build . -t gnssrefl
+      - name: Create Docker container
+        run: |
+          docker run --name gnssrefl -d -v "$PWD:/src" gnssrefl tail -f /dev/null
+      - name: Run smoke tests
+        run: |
+          docker exec -i gnssrefl /src/software_tests
+      - name: Install pytest
+        run: |
+          docker exec -i gnssrefl pip3 install 'pytest~=7.2' 'pytest-mock~=3.10'
+      - name: Run unit tests
+        run: |
+          docker exec -i gnssrefl pytest /src/test
```

Comment:
By default `pytest` will run in the current working directory, which in the gnssrefl Docker image will be the installed package. Since that does not contain any test files, we instead point it explicitly at `/src/test` which contains the unit tests from this repository.